src/omnicore/omnicore.cpp

```diff
@@ -4119,6 +4114,11 @@ int mastercore_handler_block_end(int nBlockNow, CBlockIndex const * pBlockIndex,
         const std::string& msg = strprintf("Shutting down due to failed checkpoint for block %d (hash %s)\n", nBlockNow, pBlockIndex->GetBlockHash().GetHex());
         PrintToLog(msg);
         if (!GetBoolArg("-overrideforcedshutdown", false)) AbortNode(msg, msg);
+    } else {
```

Comment:
It seems logical (the ```else``` of ```!checkpointvalid```).

Without it using ```--overrideforcedshutdown``` would still cause a block that failed checkpoint to be saved to persistence.